client/client-core/src/main/java/software/amazon/smithy/java/client/core/endpoint/EndpointImpl.java

```diff
@@ -23,8 +23,11 @@ final class EndpointImpl implements Endpoint {
 
     private EndpointImpl(Builder builder) {
         this.uri = Objects.requireNonNull(builder.uri);
-        this.authSchemes = List.copyOf(builder.authSchemes);
-        this.properties = Map.copyOf(builder.properties);
+        this.authSchemes = builder.authSchemes == null ? List.of() : builder.authSchemes;
+        this.properties = builder.properties == null ? Map.of() : builder.properties;
+        // Clear out the builder, making this class immutable and the builder still reusable.
+        builder.authSchemes = null;
+        builder.properties = null;
```

Comment:
I think this break the developer expectation regarding the state of the builder

Reply:
The builder still works fine and is reusable after building. It creates these lists and maps lazily when needed.